babel.config.js

```diff
@@ -98,8 +92,15 @@ module.exports = function getBabelConfig(api) {
   }
 
   return {
+    // https://babeljs.io/docs/assumptions
     assumptions: {
       noDocumentAll: true,
+      // With our cose these assumptions are safe, and the
+      // resulting behavior is equivalent to spec mode.
+      setPublicClassFields: true,
+      privateFieldsAsProperties: true,
+      objectRestNoSymbols: true,
+      setSpreadProperties: true,
```

Comment:
Feel free to go through https://babeljs.io/docs/assumptions and check if there is anything else you want to enable :)

Reply:
One possible effect of this PR is that `setSpreadProperties` is applied _more_, because object spread is also compiled in some cases by the destructuring transform and you didn't enable loose mode in that plugin. Now all the object spreads are transformed with this assumption, regardless of which plugin actually does it.